test/endToEnd/manualInputTest.js

```diff
@@ -0,0 +1,474 @@
+    .pause(250) // wait for debounce?
+    .click('#logo')
+    .assert.containsText('#in-progress-workspace-0', 'another title');
+}
+
+function editContext(browser) {
+  browser
+    .setValue('#therapeutic-context', NEW_THERAPEUTIC_CONTEXT)
+    .pause(250)
+    .click('#logo')
+    .click('#in-progress-workspace-0')
+    .assert.containsText('#therapeutic-context', NEW_THERAPEUTIC_CONTEXT)
+    .click('#logo');
+}
+
+function toggleFavourability(browser) {
+  browser
+    .waitForElementVisible('#favourable-criteria-label')
+    .waitForElementVisible('#unfavourable-criteria-label')
+    .waitForElementVisible('#add-favourable-criterion-cell')
+    .waitForElementVisible('#add-unfavourable-criterion-cell')
+    .click('#favourability-checkbox')
+    .assert.not.elementPresent('#favourable-criteria-label')
+    .assert.not.elementPresent('#unfavourable-criteria-label')
+    .assert.not.elementPresent('#add-favourable-criterion-cell')
+    .waitForElementVisible('#add-unfavourable-criterion-cell')
+    .click('#logo');
+}
+
+function changeCriterionFavourability(browser) {
+  const criterionTitle = 'criterion 2';
+  const favourableCriterionTitlePath = '//tbody/tr[4]/td[3]/span/span';
+  const unFavourableCriterionTitlePath = '//tbody/tr[6]/td[3]/span/span';
+
+  browser.perform(() => {
+    browser.useXpath().getAttribute('//tbody/tr[4]', 'id', (result) => {
+      var criterionId = result.value.split('-').slice(2).join('-');
+      var makeUnfavourableButton =
+        '//*[@id="make-unfavourable-' + criterionId + '"]';
+      var makeFavourableButton =
+        '//*[@id="make-favourable-' + criterionId + '"]';
+      browser
+        .waitForElementVisible(makeUnfavourableButton)
+        .click(makeUnfavourableButton)
+        .assert.containsText(unFavourableCriterionTitlePath, criterionTitle)
+        .waitForElementVisible(makeFavourableButton)
+        .click(makeFavourableButton)
+        .assert.containsText(favourableCriterionTitlePath, criterionTitle)
+        .useCss()
+        .click('#logo');
+    });
+  });
+}
+
+function addCriterion(browser) {
+  browser
+    .useXpath()
+    .click('//tbody/tr[6]/td/button')
+    .assert.containsText('//tbody/tr[6]/td[3]/span/span', 'new criterion')
+    .useCss()
+    .click('#logo');
+}
+
+function deleteCriterion(browser) {
+  browser
+    .useXpath()
+    .click('//tbody/tr[4]/td[1]/div/div[1]/button')
+    .assert.containsText('//*[@id="warnings"]/div[1]', TOO_FEW_CRITERIA_WARNING)
+    .useCss()
+    .click('#logo');
+}
+
+function moveCriterion(browser) {
+  browser.perform(() => {
+    browser.useXpath().getAttribute('//tbody/tr[2]', 'id', (result) => {
+      var criterionId = result.value.split('-').slice(2).join('-');
+      var moveDownButton = '//*[@id="move-criterion-down-' + criterionId + '"]';
+      var moveUpButton = '//*[@id="move-criterion-up-' + criterionId + '"]';
+      browser.assert
+        .containsText('//tbody/tr[2]/td[3]/span/span', 'criterion 1')
```

Comment:
might be nice to have less magic xpath strings, by putting them in variables with descriptive names